src/ICSharpCode.SharpZipLib/Zip/ZipFile.cs

```diff
@@ -1667,11 +1667,7 @@ public void Add(string fileName, CompressionMethod compressionMethod, bool useUn
 				throw new ObjectDisposedException("ZipFile");
 			}
 
-			if (!ZipEntry.IsCompressionMethodSupported(compressionMethod))
```

Comment:
This just reads out as being wrong. Why would ZipFile ask an entry if it supports compressing it?

Reply:
Yes, it seems inside out.